hw1-app/src/app/core/header/header.component.ts

```diff
@@ -1,19 +1,35 @@
-import { Component, OnInit, ChangeDetectionStrategy } from '@angular/core';
-import { IUser, User } from '../models/user.model';
+import {Component, OnInit, ChangeDetectionStrategy} from '@angular/core';
+import {IUser, User} from '../models/user.model';
+import {AuthService} from '../auth.service';
+import {Subscription} from 'rxjs';
 
 @Component({
   selector: 'header',
   templateUrl: './header.component.html',
   styleUrls: ['./header.component.scss'],
-  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class HeaderComponent implements OnInit {
-  public user: IUser = new User({ id: 'Id', firstName: 'Kate', lastName: 'Mishyna' });
+  public user: IUser = new User();
+  public authSubscription: Subscription;
 
-  constructor() {
+  constructor(private authSvc: AuthService) {
+    this.authSubscription = this.authSvc.getAuthSubject()
```

Comment:
memory leak. I see save subscription but don't see unsubscribe, but If header live while live app you don't need unsubscribe